cmd/kots/cli/install.go

```diff
@@ -643,3 +656,81 @@ func getHttpProxyEnv(v *viper.Viper) map[string]string {
 	return env
 
 }
+
+func callback(t uint32, f func() bool) func() {
+	return func() {
+		if f() {
+			time.AfterFunc(time.Duration(t)*time.Second, callback(t, f))
+		}
+	}
+}
+
+//IsAppReady checks for app readiness
+func IsAppReady(namespace string) (bool, error) {
+
+	log := logger.NewLogger()
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	clientset, err := k8sutil.GetClientset(kubernetesConfigFlags)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get clientset")
+	}
+	podName, err := k8sutil.FindKotsadm(clientset, namespace)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to find kotsadm pod")
+	}
+
+	localPort, errChan, err := k8sutil.PortForward(kubernetesConfigFlags, 0, 3000, namespace, podName, false, stopCh, log)
+	if err != nil {
+		log.FinishSpinnerWithError()
+		return false, errors.Wrap(err, "failed to start port forwarding")
+	}
+
+	go func() {
+		select {
+		case err := <-errChan:
+			if err != nil {
+				log.Error(err)
+			}
+		case <-stopCh:
+		}
+	}()
+
+	authSlug, err := auth.GetOrCreateAuthSlug(kubernetesConfigFlags, namespace)
+	if err != nil {
+		log.FinishSpinnerWithError()
+		log.Info("Unable to authenticate to the Admin Console running in the %s namespace. Ensure you have read access to secrets in this namespace and try again.", namespace)
+		os.Exit(2) // not returning error here as we don't want to show the entire stack trace to normal users
```

Comment:
this function must never terminate the process

Reply:
yes fixed.